app/design-system/design-system-hedvig/src/main/kotlin/com/hedvig/android/design/system/hedvig/RadioOption.kt

```diff
@@ -0,0 +1,562 @@
+            is IconResource.Vector -> {
+              Icon(
+                imageVector = radioOptionStyle.iconResource.imageVector,
+                contentDescription = null,
+                tint = Color.Unspecified,
+                modifier = Modifier
+                  .size(32.dp),
+              )
+            }
+
+            is IconResource.Painter -> {
+              Image(
+                painter = painterResource(id = radioOptionStyle.iconResource.painterResId),
+                contentDescription = null,
+                modifier = Modifier
+                  .size(32.dp),
+              )
+            }
+          }
+          Spacer(Modifier.width(8.dp))
+          HedvigText(
+            optionText,
+            style = radioOptionSize.size(radioOptionStyle).optionTextStyle,
+            modifier = Modifier.weight(1f),
+            color = optionTextColor,
+          )
+          Spacer(Modifier.width(8.dp))
+          SelectIndicationCircle(chosenState, lockedState)
+        }
+
+        is Label -> {
+          Column(Modifier.weight(1f)) {
+            HedvigText(
+              optionText,
+              style = radioOptionSize.size(radioOptionStyle).optionTextStyle,
+              color = optionTextColor,
+            )
+            HedvigText(
+              radioOptionStyle.labelText,
+              style = radioOptionSize.size(radioOptionStyle).labelTextStyle,
+              color = labelTextColor,
+            )
+          }
+          Spacer(Modifier.width(8.dp))
+          SelectIndicationCircle(chosenState, lockedState)
+        }
+
+        LeftAligned -> {
+          SelectIndicationCircle(chosenState, lockedState)
+          Spacer(Modifier.width(8.dp))
+          HedvigText(
+            optionText,
+            style = radioOptionSize.size(radioOptionStyle).optionTextStyle,
+            color = optionTextColor,
+            modifier = Modifier.weight(1f),
+          )
+        }
+      }
+    }
+  }
+}
+
+@Composable
+internal fun SelectIndicationCircle(
+  chosenState: ChosenState,
+  lockedState: LockedState,
+  modifier: Modifier = Modifier,
+) {
+  CheckItemAnimation(chosenState) { currentState: ChosenState ->
+    Box(
+      contentAlignment = Alignment.Center,
+    ) {
+      Spacer(
+        modifier
+          .size(24.dp)
+          .clip(CircleShape)
+          .then(
+            when (currentState) {
+              Chosen -> {
+                when (lockedState) {
+                  Locked -> Modifier.border(8.dp, radioOptionColors.disabledIndicatorColor, CircleShape)
+                  NotLocked -> Modifier.border(8.dp, radioOptionColors.chosenIndicatorColor, CircleShape)
+                }
+              }
+
+              NotChosen -> {
+                Modifier.border(2.dp, radioOptionColors.notChosenIndicatorColor, CircleShape)
+                // disabled color is actually brighter than notChosenIndicatorColor,
+                // makes no sense here, so I left only one
```

Comment:
Did the design purposefully chose it to be that way?
Does it look confusing if we have a disabled and a non-disabled but unselected option side by side?
Could it be a bit confusing if perhaps in the future the color for disabled is adjusted, but the not-selected is not, and we make the change in the tokens but the implementation changes for the not-selected option too, which might be a bit confusing say 1 year in the future.

Or is the part that does not add up something that should be brought up with design perhaps?

Reply:
_Does it look confusing if we have a disabled and a non-disabled but unselected option side by side?_ - yes, basically this. Maybe it won't ever be the case, or very rarely, so maybe I would just put notChosenIndicatorColor there